File: CarDealershipAPI/CarDealershipAPI/Models/Car.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Web;

namespace CarDealershipAPI.Models
{
    public class Car
    {
        public int CarID { get; set; }

        [Required]
        public string CarNumberPlate { get; set; }

        [Required]
        public int? CarDealershipID { get; set; }

        [Required]
        public int? BrandID { get; set; }

        [Required]
        public int? Kilometers { get; set; }
    }
}

File: CarDealershipApp/src/app/models/car.ts
import { Brand } from './brand';
import { CarDealership } from './carDealership';

/**
 * Car model
 * 
 * @export
 * @class Car
 * @author pedro.calero
 */
export class Car {
  CarID: number;
  CarNumberPlate: String;
  CarDealershipID: number;
  BrandID: number;
  Kilometers: number;
  CarDealership: CarDealership;
  Brand: Brand;

  /**
   * Creates an instance of Car.
   * @memberof Car
   */
  constructor (){
  }

}